src/coreclr/gc/gc.cpp

```diff
@@ -3930,6 +3920,43 @@ void region_allocator::delete_region (uint8_t* region_start)
 
     leave_spin_lock();
 }
+
+void region_allocator::move_highest_free_regions (int64_t n, bool small_region_p, region_free_list to_free_list[count_free_region_kinds])
+{
+    assert (n > 0);
+
+    uint32_t* current_index = region_map_left_end - 1;
+    uint32_t* lowest_index = region_map_left_start;
+
+    while (current_index >= lowest_index)
+    {
+        uint32_t current_val = *current_index;
+        uint32_t current_num_units = get_num_units (current_val);
+        bool free_p = is_unit_memory_free (current_val);
+        if (!free_p && ((current_num_units == 1) == small_region_p))
+        {
+            uint32_t* index = current_index - (current_num_units - 1);
+            heap_segment* region = get_region_info (region_address_of (index));
+            if (heap_segment_allocated (region) == nullptr)
+            {
```

Comment:
I would make a global func called `is_free_region` that encapsualtes this logic in case we change how we represent a free region in the future

Reply:
Good suggestion - done.